### PR TITLE
"Behind the proxy" support for devlab setup

### PR DESCRIPTION
**Pull request by raiesmh08**
As per current design of Cloudferry, Vagrantfile does not considers
'host behind proxy' problem.

If one is trying to setup Cloudferry devlabs, booted VM will have no Internet
access, as Vagrantfile does consider proxy considerations.

Again, with no internet access it not be able to downloads required packages.

When I tried 'vagrant up juno', devstack was not able to download.
And this was basically because of lack of support of 'behind the proxy'.

This patch eliminates this problem.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
MirantisWorkloadMobility

**Comment by MirantisWorkloadMobility** [comment]
Pylint check succeeded

**Comment by MirantisWorkloadMobility** [comment]
MirantisWorkloadMobility

**Comment by MirantisWorkloadMobility** [comment]
MirantisWorkloadMobility

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1396/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1395: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1405: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1407: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1408: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1409: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1410: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1411: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1412: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:81: [W1202(logging-format-interpolation), NeutronNetwork.read_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/network/neutron.py:149: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:259: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 21)
cloudferrylib/os/network/neutron.py:727: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:831: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:856: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:887: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:948: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1190: [W1201(logging-not-lazy), NeutronNetwork.upload_routers] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1207: [W0101(unreachable), NeutronNetwork.add_router_interfaces] Unreachable code
cloudferrylib/os/network/neutron.py:1327: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1334: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1338: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1406/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1407/

**Comment by MirantisWorkloadMobility** [comment]
Pylint check succeeded

**Comment by roman-verchikov** [comment]
Please remove unnecessary commits from this pull request

**Comment by MirantisWorkloadMobility** [comment]
Pylint check succeeded

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1427/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Icehouse to Juno migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-icehouse-to-juno/431/

**Comment by MirantisWorkloadMobility** [comment]
Ready

**Comment by mgrygoriev** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
There were no python files updated. Admins, please, check pull request.

File: devlab/Vagrantfile [hunk 1]
```diff
@@ -42,7 +42,21 @@ nodes = {
...
+    end
+end
+
 Vagrant.configure(2) do |config|
+
+  if Vagrant.has_plugin?("vagrant-proxyconf")
+    config.proxy.http     = "#{options['http_proxy']}"
+    config.proxy.https    = "#{options['https_proxy']}"
+    config.proxy.https    = "#{options['ftp_proxy']}"
+    config.proxy.no_proxy = "localhost,127.0.0.1,#{options['grizzly_ip']},#{options['icehouse_ip']},#{options['juno_ip']},#{options['nfs_ip']}"
```
Comment (mgrygoriev) [inline devlab/Vagrantfile]: Please, rebase and add cloudferry_ip as well.

Comment (raiesmh08) [inline devlab/Vagrantfile]: Hi mgrygoriev, 

Do we really nead cloudferry_ip ? 
In devlab environment, as we know, we used to make only 4 VMs (grizzly, icehouse, juno, and nfs). We do not make cloudferry VM. Instead we used to make cloudferry on our host machine. I think in this case we do not need to add cloudferry_ip in no_proxy.

Comment (roman-verchikov) [inline devlab/Vagrantfile]: `cloudferry` VM has been restored recently based on multiple requests from folks trying to run migration on windows machines.

Comment (roman-verchikov) [inline devlab/Vagrantfile]: Also please keep the line length at 120 symbols, it's impossible to read such long lines on github

Comment (raiesmh08) [inline devlab/Vagrantfile]: Done in new pull request - https://github.com/MirantisWorkloadMobility/CloudFerry/pull/635

Unfortunately, my local fork and repo was deleted. Thus I was unable to recover my branch. 
Thanks why putting separated pull request for this.

**Comment by roman-verchikov** [comment]
Continued in https://github.com/MirantisWorkloadMobility/CloudFerry/pull/635

**Pull request closed without merging**